temporal-testing/src/main/java/io/temporal/internal/testservice/TestWorkflowStoreImpl.java

```diff
@@ -92,19 +93,17 @@ void addAllLocked(List<HistoryEvent> events, Timestamp eventTime) {
       for (HistoryEvent event : events) {
         HistoryEvent.Builder eBuilder = event.toBuilder();
         if (completed) {
-          throw Status.FAILED_PRECONDITION
-              .withDescription(
-                  "Attempt to add an eBuilder after a completion eBuilder: "
-                      + WorkflowExecutionUtils.prettyPrintObject(eBuilder))
-              .asRuntimeException();
+          throw ApplicationFailure.newNonRetryableFailure("Workflow execution completed.", "");
         }
         eBuilder.setEventId(history.size() + 1L);
         // It can be set in StateMachines.startActivityTask
         if (Timestamps.toMillis(eBuilder.getEventTime()) == 0) {
           eBuilder.setEventTime(eventTime);
         }
         history.add(eBuilder.build());
-        completed = completed || WorkflowExecutionUtils.isWorkflowExecutionCompletedEvent(eBuilder);
+        if (WorkflowExecutionUtils.isWorkflowExecutionCompletedEvent(eBuilder)) {
```

Comment:
why do we need this change? The code to the left is more canonical and easy to read.

Reply:
Hm, I actually changed it because to me this version is more readable and straight forward. It very clearly states the only condition when completed becomes true. :)